packages/integration-karma/test/mixed-shadow-mode/dual-component/index.spec.js

```diff
@@ -0,0 +1,57 @@
+import { createElement, setFeatureFlagForTest } from 'lwc';
+import { extractDataIds } from 'test-utils';
+import Container from 'x/container';
+
+// compat will have the token when rendering in native
+if (!process.env.NATIVE_SHADOW && !process.env.COMPAT) {
+    describe('dual component', () => {
+        beforeEach(() => {
+            setFeatureFlagForTest('ENABLE_MIXED_SHADOW_MODE', true);
+        });
+        afterEach(() => {
+            setFeatureFlagForTest('ENABLE_MIXED_SHADOW_MODE', false);
+        });
+
+        ['native', 'synthetic'].forEach((firstRenderMode) => {
+            describe(`renders first in ${firstRenderMode}`, () => {
+                it(`should set the tokens for synthetic shadow`, () => {
+                    const elm = createElement('x-container', { is: Container });
+                    elm.syntheticFirst = firstRenderMode === 'synthetic';
+                    document.body.appendChild(elm);
+
+                    const synthetic = extractDataIds(elm.shadowRoot.querySelector('x-component'));
+                    const native = extractDataIds(
+                        elm.shadowRoot
+                            .querySelector('x-native')
+                            .shadowRoot.querySelector('x-component')
+                    );
+
+                    expect(synthetic.div.hasAttribute('x-component_component')).toBe(true);
+                    expect(native.div.hasAttribute('x-component_component')).toBe(false);
+                });
+
+                it(`should mangle the IDs in synthetic shadow`, () => {
+                    const elm = createElement('x-container', { is: Container });
+                    elm.syntheticFirst = firstRenderMode === 'synthetic';
+                    document.body.appendChild(elm);
+
+                    const synthetic = extractDataIds(elm.shadowRoot.querySelector('x-component'));
+                    const native = extractDataIds(
+                        elm.shadowRoot
+                            .querySelector('x-native')
+                            .shadowRoot.querySelector('x-component')
+                    );
+
+                    expect(synthetic.label.id).toMatch(/^foo-/);
+                    expect(synthetic.input.getAttribute('aria-labelledby')).toMatch(/^foo-/);
+                    expect(synthetic.label.id).toEqual(
+                        synthetic.input.getAttribute('aria-labelledby')
+                    );
+
+                    expect(native.label.id).toEqual('foo');
+                    expect(native.input.getAttribute('aria-labelledby')).toEqual('foo');
+                });
+            });
```

Comment:
The style token attributes and ID scoping are just two visible effects of `shadowMode` that we can test here. I don't think the ordering actually matters, but I kept it from [Jose's other test](https://github.com/salesforce/lwc/pull/2781/files#diff-94dd7a7a34764ea09cb3ead76a352da2c31cf1f36b9019df5ed72ee9ea936331).